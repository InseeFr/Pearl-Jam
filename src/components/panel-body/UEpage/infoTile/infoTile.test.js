import { MemoryRouter, Route } from 'react-router';

import InfoTile from './infoTile';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { render } from '@testing-library/react';

jest.mock('indexedbb/services/surveyUnit-idb-service', () => {
  const originalModule = jest.requireActual('indexedbb/services/surveyUnit-idb-service');
  const suIdbService = originalModule.default;
  suIdbService.getById = async () => {
    return Promise.resolve({
      persons: [{ firstName: 'FirstName', lastName: 'LastName', privileged: true }],
      campaign: 'Mocked campaign',
      sampleIdentifiers: { ssech: '1' },
    });
  };
  originalModule.default = suIdbService;
  return {
    __esModule: true,
    ...originalModule,
  };
});

const SuIdbService = require('indexedbb/services/surveyUnit-idb-service');

const MemoryRouterWithInitialRoutes = ({ children }) => (
  <MemoryRouter initialEntries={['/survey-unit/1']}>
    <Route path="/survey-unit/:id">{children}</Route>
  </MemoryRouter>
);

const customRender = ui => {
  // return render(ui, { wrapper: MemoryRouterWithInitialRoutes });
  return render(ui, { wrapper: MemoryRouterWithInitialRoutes });
};

const wait = async () => new Promise(resolve => setTimeout(resolve, 0));

it('renders correctly', async () => {
  const result = customRender(<InfoTile refresh={true} />);
  await act(async () => {
    await wait();
  });
  expect(result.baseElement).toMatchSnapshot();

  const { getById } = SuIdbService.default;
  // check if mock is done
  await expect(getById(5)).resolves.toStrictEqual({
    persons: [{ firstName: 'FirstName', lastName: 'LastName', privileged: true }],
    campaign: 'Mocked campaign',
    sampleIdentifiers: { ssech: '1' },
  });
});
