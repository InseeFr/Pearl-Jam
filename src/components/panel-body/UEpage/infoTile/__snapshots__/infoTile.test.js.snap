// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation0 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root makeStyles-column-3"
      >
        <div
          class="makeStyles-row-2"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
          <p
            class="MuiTypography-root makeStyles-bold-7 makeStyles-overflow-8 MuiTypography-body1"
          >
            FirstName LastName
          </p>
        </div>
        <div
          class="makeStyles-row-2"
        >
          <p
            class="MuiTypography-root makeStyles-rounded-6 makeStyles-title-4 MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-alignCenter"
          >
            1
          </p>
          <p
            class="MuiTypography-root makeStyles-title-4 makeStyles-overflow-8 MuiTypography-body1 MuiTypography-colorTextSecondary"
          >
            Mocked campaign
          </p>
        </div>
        <h6
          class="MuiTypography-root makeStyles-title-4 makeStyles-bold-7 MuiTypography-h6 MuiTypography-colorError"
        >
          # 1
        </h6>
      </div>
    </div>
  </div>
</body>
`;
